### DISCUSSION TITLE
This confirmation is totally unnecesary and real overkill! Please remove this step!

### DISCUSSION THREAD

**Discussion by bencresty** [craftcms/cms#11573] [Ideas] [answered]
Not sure how this is in Craft 4, but in Craft 3 when removing panels from the Dashboard after clicking the cross to remove it, we also get an alert as confirmation.

When needing to remove all panels from each new installation this is wayyyyy to overkill to me. We'd like to just remove all panels by clicking first, second, third and fifth cross-buttons in the dashboard settings. But now we need to do all kinds of extra clicks as if it's something huge what's happening here. While we can easily re-enable panels in the same settings, even if we would accidentally remove one.

![image](https://user-images.githubusercontent.com/69322733/177854338-a07598a5-f622-440a-b95c-a0a4370fbc00.png)

Please, remove this way over the top alert-box! It doesn't make any sense to us to have this extra confirmation for such a small action. It should be easy for you to remove this alert in code, but the outcome will be a way cleaner looking, easier UI with less steps!

Thanks in advance!

**Comment by MattWilcox** [comment]
I hardly ever use dashboard widgets, but imagine this gets old fast if you're doing bulk editing. It's a good protection from mis-clicks though.

1) Can only Admin's delete widgets?
2) If so, then maybe you only need the initial warning, and the messaging could become "Are you sure you want to delete? [Yes, and suppress further warnings for 5min] [Cancel]

?

**Reply by brandonkelly** [reply]
Each user has their own unique set of dashboard widgets, which they are in full control over.

**Reply by bencresty** [reply]
> I hardly ever use dashboard widgets

Me neither, except for projects where we create them ourselves, that's why I always want them removed for projects

> It's a good protection from mis-clicks though.

I don't agree at all. The thing is, nobody needs a conformation on this harmless action. Because:

1) The chance that anybody ever removes a panel by misclicking is practically 0.0000,,,1% as the user needs to open the menu first and then also hit the small cross
2) In the extreme case a user does remove a panel by accident it takes two clicks and 2 seconds to bring it back via the menu next to it, so it's no issue at all if that happens. 
3) In all other 99.99999-something % users hit a cross to remove a panel they do this on purpose and then it's pretty annoying to get an alert box. It also feels patronizing if the CMS doesn't trust you and warns you on such a complete harmless action
4) With adding this extra confirmation step (that AFAIK never happens anywhere else in the CMS) you put a lot of extra weight on the harmless action causing users to think and sensible users to even get scared about what they're about to do, while that's completely out of place as nothing big destructive is happening. It's just hiding a panel basically.
5) Javascript alerts never fits any layout and are always a bad decision IMO

While if the alert wasn't ther the user would be much happier using the interface on all 99.99999-something % other times, because she/he/it feels trusted by the CMS, the CMS feels intuitive and responsive and it/he/she/the can quickly remove all panels without any redundant confirmations.


> maybe you only need the initial warning, and the messaging could become "Are you sure you want to delete? [Yes, and suppress further warnings for 5min] [Cancel]

Yeah, that would make the interface even weirder. Please for the love of intuitive and responsive interfaces: no!!

Crazy thing is that we can remove all kinds of things that ARE destructive and painful if it happens in the CP without confirmation (and please keep it that way!!), but the most harmless thing in the CMS that's never causing any harm as it's always non-destructive gets this ugly overly worried alert box... it just doesn't make sense.


> Each user has their own unique set of dashboard widgets, which they are in full control over.

Not sure what you want to add with this? The discussion isn't about the fact that users might want to remove widgets or not, it's only about that extra redundant confirmation step.

Bottom line: 
If a user hits a cross to remove a panel it wants to... remove a panel. To confirm that would be a useless extra step which is only annoying. In the extreme rare case the user made a mistake it has the panel back in 2 seconds.

**Reply by MattWilcox** [reply]
Ahhh, yeah I forgot that. You can't set them up for a client without logging in as the client - because they're personal.

Well, given that they are personal to create and delete maybe:
```
"Are you sure you want to delete? 
[Yes] [Yes, and suppress further warnings for 5min] [Cancel]
```

**Reply by bencresty** [reply]
@MattWilcox Sorry, either you didn't read my post fully, or you didn't get what I wrote at all.

BTW scary people seeing this alert would be even more scared:
![image](https://user-images.githubusercontent.com/69322733/178015555-13f84b42-b504-4115-a7e2-ab90c87ec16e.png)
Because what does this even mean to them? The message doesn't even tell WHAT will be removed...

**Reply by AugustMiller** [reply]
This last screenshot looks like an actual bug with the Support widget not having all the recommended/required title/label properties defined… i.e. "Craft News (Feed)" is has a configured "name" of `Craft News`, and it's a `Feed` widget, but the `Craft Support` widget instance doesn't have a default name.

**Reply by brandonkelly** [reply]
Just fixed that `“”` bug for the next Craft 3 and 4 releases (86f443687a6e6250b39c393e982aeb2fedea309e).

**Reply by MattWilcox** [reply]
> Sorry, either you didn't read my post fully, or you didn't get what I wrote at all

We posted at the same time.

**Reply by brandonkelly** [reply]
3.7.48 and 4.1.4 are both out with the `“”` fix.

**Comment by brandonkelly** [comment, marked as answer]
We discussed this a bit in Discord, and the consensus was that the confirmation dialog should remain, since it’s an immediately destructive action.

While I realize this falls a bit into the “tech savvy” territory, it’s possible to quickly delete multiple widgets, even with the confirmation dialogs, using keyboard controls.

There was a suggestion to make it easier to “reset” the Dashboard to a blank slate, without needing to delete each widget individually. While that could help this specific situation, I’m not sure it’s a common enough problem to warrant the additional UI.

Instead, we are going to look into improving the flash notice UI so that they can become interactive by including links or buttons, etc. In this case, that could mean adding an “Undo” button to the `[Widget name] deleted` notice, which would create a new dashboard widget with the same configuration as the one just deleted.

**Reply by bencresty** [reply]
Sorry for the late response. Guess I've missed this one. Sounds like a great solution to me. Thanks!

**Reply by brandonkelly** [reply]
Just PR’d that change for Craft 4.2 yesterday :) #11612

**Reply by bencresty** [reply]
@brandonkelly  Thanks. And ooooh, when quickly reading it I like this:
![image](https://user-images.githubusercontent.com/69322733/179554320-88c58104-e90a-420f-b6a3-6fc557411e76.png)
and this:
![image](https://user-images.githubusercontent.com/69322733/179554444-feab7296-5b8e-4ae3-acca-1d9d3ff4987b.png)
A loooot! That is definitely an improvement.

(Just a thing to keep in mind if you target mobile and touch driven devices too: on mobile devices and touch there's no such thing as hover)

Great to see these changes and thanks for taking this seriously! Very appreciated

**Reply by brandonkelly** [reply]
> (Just a thing to keep in mind if you target mobile and touch driven devices too: on mobile devices and touch there's no such thing as hover)

Yeah that specific feature doesn’t apply to mobile, but if you tap anywhere in the notification, it will be treated the same as a click on a desktop browser, and keep the notification visible indefinitely, until you tap the close button.

**Reply by brandonkelly** [reply]
Craft 4.2.0 is out now with those changes ✨
